docs/src/content/hardhat-runner/docs/guides/test-contracts.md

```diff
@@ -1,3 +1,197 @@
+  const ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60;
+  unlockTime = (await helpers.time.latest()) + ONE_YEAR_IN_SECS;
+
+  const Lock = await ethers.getContractFactory("Lock");
+  lock = await Lock.deploy(unlockTime, { value: lockedAmount });
+});
+
+it("some test", async function () {
+  // use the deployed contract
+});
+```
+
+There are two problems with this approach:
+
+- If you have to deploy many contracts, your tests will be slower because each one has to send multiple transactions as part of its setup.
+- Sharing the variables like this between the `beforeEach` hook and your tests is ugly and error-prone.
+
+The `loadFixture` helper in the Hardhat Network Helpers fixes both of these problems. This helper receives a _fixture_, a function that sets up the chain to some desired state. The first time `loadFixture` is called, the fixture is executed. But the second time, instead of executing the fixture again, `loadFixture` will reset the state of the network to the point where it was right after the fixture was executed. This is faster, and it undoes any state changes done by the previous test.
+
+This is how our two first tests look like when a fixture is used:
+
+```tsx
+describe("Lock", function () {
+  async function deployOneYearLockFixture() {
+    const lockedAmount = 1_000_000_000;
+    const ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60;
+    const unlockTime = (await time.latest()) + ONE_YEAR_IN_SECS;
+
+    const Lock = await ethers.getContractFactory("Lock");
+    const lock = await Lock.deploy(unlockTime, { value: lockedAmount });
+
+    return { lock, unlockTime, lockedAmount };
+  }
+
+  it("Should set the right unlockTime", async function () {
+    const { lock, unlockTime } = await loadFixture(deployOneYearLockFixture);
+
+    // assert that the value is correct
+    expect(await lock.unlockTime()).to.equal(unlockTime);
+  });
+
+  it("Should revert with the right error if called too soon", async function () {
+    const { lock } = await loadFixture(deployOneYearLockFixture);
+
+    await expect(lock.withdraw()).to.be.revertedWith("You can't withdraw yet");
+  });
+});
+```
+
+The fixture function can return anything you want, and the `loadFixture` helper will return it. We recommend returning an object like we did here, so you can extract only the values you care about for that test.
+
+### Other tests
+
+There are several other things you can test, like [which events are emitted](/hardhat-chai-matchers/docs/reference#.emit) or [how the balances of the involved addresses change](/hardhat-chai-matchers/docs/reference#balance-change). Check the `test/Lock.js` file to see the other examples.
+
+### Running tests in parallel
+
+You can run your tests in parallel by using the `--parallel` flag:
+
+```
+npx hardhat test --parallel
+```
+
+Alternatively, use `parallel: true` in the `mocha` section of your Hardhat config, which will enable parallelism for plugins that don't have a `--parallel` flag, like `solidity-coverage`.
```

Comment:
I think this paragraph creates more confusion than anything else.

Reply:
What about just removing everything after "which will enable..."?

Reply:
I'm ok with just deleting the whole paragraph though.

Reply:
> What about removing everything after "which will enable..."?

Yeah, that works